packages/amazon/src/serverGroup/details/scalingPolicy/upsert/ScalingPolicyCommandBuilderService.ts

```diff
@@ -0,0 +1,127 @@
+import { cloneDeep } from 'lodash';
+
+import {
+  ISimplePolicyDescription,
+  IStepPolicyDescription,
+  IUpsertAlarmDescription,
+  IUpsertScalingPolicyCommand,
+} from '../ScalingPolicyWriter';
+import { IAmazonServerGroup, IScalingPolicy, IStepAdjustment, ITargetTrackingPolicy } from '../../../../domain';
+
+type PolicyType = 'Step' | 'TargetTracking';
+
+export const ScalingPolicyCommandBuilder = {
+  buildAlarm: (policy: IScalingPolicy, region: string, asgName: string): IUpsertAlarmDescription => {
+    const alarm = policy?.alarms[0];
+    return {
+      name: alarm.alarmName,
+      actionsEnabled: true,
+      alarmActionArns: alarm.alarmActions,
+      alarmDescription: alarm.alarmDescription,
+      asgName: asgName,
+      comparisonOperator: alarm.comparisonOperator,
+      dimensions: alarm.dimensions,
+      evaluationPeriods: alarm.evaluationPeriods,
+      insufficientDataActionArns: alarm.insufficientDataActions,
+      metricName: alarm.metricName,
+      namespace: alarm.namespace,
+      okActionArns: alarm.okactions,
+      period: alarm.period,
+      region,
+      statistic: alarm.statistic,
+      threshold: alarm.threshold,
+      unit: alarm.unit,
+    };
+  },
+
+  buildStepPolicy: (policy: IScalingPolicy, threshold: number, cooldown: number): IStepPolicyDescription => {
+    const stepAdjustments = policy.stepAdjustments.map((adjustment) => {
+      const step = {
+        scalingAdjustment: Math.abs(adjustment.scalingAdjustment),
+      } as IStepAdjustment;
+
+      if (adjustment.metricIntervalUpperBound !== undefined) {
+        step.metricIntervalUpperBound = adjustment.metricIntervalUpperBound + threshold;
+      }
+      if (adjustment.metricIntervalLowerBound !== undefined) {
+        step.metricIntervalLowerBound = adjustment.metricIntervalLowerBound + threshold;
+      }
+
+      return step;
+    });
+
+    return {
+      estimatedInstanceWarmup: policy.estimatedInstanceWarmup || cooldown || 600,
```

Comment:
```suggestion
      estimatedInstanceWarmup: policy.estimatedInstanceWarmup ?? cooldown ?? 600,
```
Should generally prefer `??` over `||` especially when working with numbers so explicitly set `0` values don't get overwritten by a default value.